### PR TITLE
API-24186 - Un-revert VBADocuments max page limit, fix upload error detail

### PR DESCRIPTION
**Pull request by caseywilliams**
## Summary

- I had previously added these changes in #12457, but I had to revert it because the error details for DOC106 and DOC108 were not being saved/recorded in the Benefits Intake error report (or in the database)
- This PR un-reverts the changes, and also fixes (as far as I can tell) the missing error details (see the second commit here for the fix)

## Testing done

- Updated automated tests to confirm the error detail is set

## What areas of the site does it impact?
- The maximum page size limit used in Benefits Intake will not change immediately, but it will be increased to 79x101 if we set the `vba_documents_larger_page_size_limit`
- This updates the UploadError handling a little, but should not change the actual behavior immediately

## Acceptance criteria

- [X]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [X]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [X]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

### CODE REVIEW

File: modules/vba_documents/lib/vba_documents/upload_validator.rb [hunk 1]
```diff
@@ -112,32 +112,28 @@ def validate_line_of_business(lob, submission_version)
...
+      result = PDFValidator::Validator.new(file_path, pdf_validator_options).validate
 
       unless result.valid_pdf?
         errors = result.errors
 
         if errors.grep(/#{PDFValidator::FILE_SIZE_LIMIT_EXCEEDED_MSG}/).any?
-          raise VBADocuments::UploadError.new(code: 'DOC106',
-                                              detail: 'Maximum document size exceeded. Limit is 100MB per document')
+          code = 'DOC106'
+          detail = CentralMail::UploadError.default_message(code, pdf_validator_options)
```
Comment (caseywilliams) [inline modules/vba_documents/lib/vba_documents/upload_validator.rb:123]: I feel silly for this, but as far as I can see, this was the center of the missing error details - I guess I hadn't noticed that we were explicitly setting the `detail` here (vs. the `message`) before. Even though the UploadError was setting a message, my previous updates neglected to keep the `detail` intact. As far as I can tell, the behavior now matches the old behavior - allowing `CentralMail::UploadError` to set the `message`, but also setting the `detail` here.

File: lib/central_mail/upload_error.rb [hunk 2]
```diff
@@ -23,15 +26,27 @@ class UploadError < StandardError
...
+      case code.to_s
+      when 'DOC106'
+        "#{message} Limit is #{PDFUtilities.formatted_file_size(opts[:size_limit_in_bytes])} per document."
+      when 'DOC108'
+        "#{message} Limit is #{opts[:width_limit_in_inches]} in x #{opts[:height_limit_in_inches]} in."
+      else
+        message
       end
-      super(message)
+    end
```
Comment (kristen-brown) [inline lib/central_mail/upload_error.rb:46]: 👍

**Review by kristen-brown — APPROVED** [review]

**Comment by kristen-brown** [comment]
@caseywilliams I forgot to comment that I think it would be a good idea to update any DOC106/108 records that were missing the error `detail`, once this PR has been deployed.

File: modules/vba_documents/lib/vba_documents/upload_validator.rb (hunk 1 above)
Comment (kristen-brown) [reply modules/vba_documents/lib/vba_documents/upload_validator.rb:123]: This is just a suggestion, and I'm not sure if I feel it's actually better, but structuring the code the following way would remove the need for `default_message` and for the second call to the `CentralMail::UploadError` class.

```
upload_error = VBADocuments::UploadError.new(code: 'DOC106', detail: 'Maximum document size exceeded. Limit is 100MB per document')
code = upload_error.code
detail = upload_error.detail

raise upload_error
```

**Pull request merged**